### PR TITLE
Always use the RedisAPI

### PR DESCRIPTION
**Pull request by dominik-cnx**
With this PR we remove below method and always use the RedisAPI

RedisProvider

Future<Redis> connection();

### CODE REVIEW

File: src/main/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandler.java [hunk 1]
```diff
@@ -997,7 +997,7 @@ private void getQueuesStatistics(RoutingContext ctx) {
                         }
                     } else {
                         String error = "Error gathering names of active queues";
-                        log.error(error);
+                        log.error(error, reply.cause());
```
Comment (mcweba) [inline src/main/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandler.java:1000]: Make use of new _RedisQuesExceptionFactory_ when logging stacktraces.

**Review by mcweba — CHANGES_REQUESTED** [review]

**Comment by hiddenalpha** [comment]
> With this PR we remove below method and always use the RedisAPI

(*Just curious*) How do those two differ?

**Review by hiddenalpha — APPROVED** [review]

**Comment by dominik-cnx** [comment]
> > With this PR we remove below method and always use the RedisAPI
> 
> (_Just curious_) How do those two differ?

The API holds the connection internally. I'm not entirely sure if we shouldn't use the connection directly, but going through the API seems more like the way to go.


public class RedisAPIImpl implements RedisAPI {

  private static final Logger LOG = LoggerFactory.getLogger(RedisAPIImpl.class);

  private final Redis redis;
  private final RedisConnection connection;

  public RedisAPIImpl(RedisConnection connection) {
    this.connection = connection;
    this.redis = null;
  }

**Comment by hiddenalpha** [comment]
> The API holds the connection internally. I'm not entirely sure if we shouldn't use the connection directly, but going through the API seems more like the way to go.

Just a thought, could this relate to #165? So maybe we just fixed it?

**Pull request merged**